crates/router/src/types/domain/user.rs

```diff
@@ -158,6 +158,40 @@ pub struct UserPassword(Secret<String>);
 impl UserPassword {
     pub fn new(password: Secret<String>) -> UserResult<Self> {
         let password = password.expose();
+
+        let mut has_upper_case = false;
+        let mut has_lower_case = false;
+        let mut has_numeric_value = false;
+        let mut has_special_character = false;
+        let mut has_whitespace = false;
+
+        for c in password.chars() {
+            has_upper_case = has_upper_case || c.is_uppercase();
+            has_lower_case = has_lower_case || c.is_lowercase();
+            has_numeric_value = has_numeric_value || c.is_numeric();
+            has_special_character =
+                has_special_character || !(c.is_alphanumeric() && c.is_whitespace());
+            has_whitespace = has_whitespace || c.is_whitespace();
+        }
+
+        let is_password_format_valid = has_upper_case
+            && has_lower_case
+            && has_numeric_value
+            && has_special_character
+            && !has_whitespace;
+
+        let is_too_long = password.graphemes(true).count() > consts::user::MAX_PASSWORD_LENGTH;
+        let is_too_short = password.graphemes(true).count() < consts::user::MIN_PASSWORD_LENGTH;
+
+        if is_too_short || is_too_long || !is_password_format_valid {
+            Err(UserErrors::PasswordParsingError.into())
+        } else {
+            Ok(Self(password.into()))
+        }
+    }
+
+    pub fn new_without_validation(password: Secret<String>) -> UserResult<Self> {
```

Comment:
may be `new_password_without_validation`,
we are not validating the password in merchant create request?

Reply:
We're retrieving user details from the database to create a new merchant. It's possible that some old users didn't followed the password convention. Therefore, we've implemented a workaround specifically for this scenario to bypass validation.